beacon_node/network/src/sync/block_lookups/single_block_lookup.rs

```diff
@@ -171,7 +171,10 @@ impl<T: BeaconChainTypes> SingleBlockLookup<T> {
         self.awaiting_parent.is_some()
             || self.block_request_state.state.is_awaiting_event()
             || match &self.component_requests {
-                ComponentRequests::WaitingForBlock => true,
+                // If components are waiting for the block request to complete, here we should
+                // check if the`block_request_state.state.is_awaiting_event(). However we already
+                // checked that above, so `WaitingForBlock => false` is equivalent.
+                ComponentRequests::WaitingForBlock => false,
```

Comment:
I think this makes sense.

I wonder if this would drop any blocks that are awaiting processing?
https://github.com/sigp/lighthouse/blob/886d409b858a63f6a811abd9cf60e2a4d6024bc2/beacon_node/network/src/sync/block_lookups/single_block_lookup.rs#L500

Reply:
This is unlikely an issue - if we do need the block, external events should somehow keep this lookup in place, a single disconnected peer should not affect this.